client-core/src/transaction_builder/default_transaction_builder.rs

```diff
@@ -46,67 +47,62 @@ where
-        Self {
-            signer,
-            fee_algorithm,
-            transaction_obfuscation,
-        }
-    }
-}
-
-impl<S, F, O> TransactionBuilder for DefaultTransactionBuilder<S, F, O>
-where
-    S: Signer,
-    F: FeeAlgorithm,
-    O: TransactionObfuscation,
-{
-    fn build(
+    ///  Create a `DummySigner` which signs a transaction with dummy values for fees calculation.
+    /// Returns a result of fees , `Tx` and selected unspent transactions
+    pub fn get_fees<'a>(
         &self,
-        name: &str,
-        passphrase: &SecUtf8,
+        total_pubkeys_len: usize,
         outputs: Vec<TxOut>,
         attributes: TxAttributes,
-        unspent_transactions: UnspentTransactions,
         return_address: ExtendedAddr,
-    ) -> Result<TxAux> {
+        unspent_transactions: &'a UnspentTransactions,
+    ) -> Result<(Coin, Tx, SelectedUnspentTransactions<'a>)> {
         let output_value = sum_coins(outputs.iter().map(|output| output.value)).chain(|| {
             (
                 ErrorKind::IllegalInput,
                 "Sum of output values exceeds maximum allowed amount",
             )
         })?;
         let mut fees = Coin::zero();
-
-        loop {
-            let (selected_unspent_transactions, difference_amount) =
+        let dummy_signer = DummySigner {};
+        let (tx, selected_unspent_txs) = loop {
+            let (selected_unspent_txs, difference_amount) =
                 unspent_transactions.select((output_value + fees).chain(|| {
                     (
                         ErrorKind::IllegalInput,
                         "Sum of output values and fee exceeds maximum allowed amount",
                     )
                 })?)?;
 
-            let transaction = build_transaction(
-                &selected_unspent_transactions,
+            let tx = build_transaction(
+                &selected_unspent_txs,
                 outputs.clone(),
                 attributes.clone(),
                 difference_amount,
                 return_address.clone(),
             );
 
-            let witness = self.signer.sign(
-                name,
-                passphrase,
-                transaction.id(),
-                selected_unspent_transactions,
-            )?;
-
-            let signed_transaction = SignedTransaction::TransferTransaction(transaction, witness);
-            let tx_aux = self.transaction_obfuscation.encrypt(signed_transaction)?;
-
+            // use the dummy signer to sign the selected unspent transactions
+            let witness = dummy_signer.sign(total_pubkeys_len, &selected_unspent_txs)?;
+            let plain_payload_len = PlainTxAux::TransferTx(tx.clone(), witness).encode().len();
+            //  pad the payload to the multiples of 128bit(8bit*16)
+            let padded_payload = if plain_payload_len % 16_usize == 0 {
+                vec![0; plain_payload_len]
+            } else {
+                vec![0; plain_payload_len + 16_usize]
+            };
```

Comment:
how about
asserting
assert!(0== padded_payload.len()%16);

Reply:
why asserting?

Reply:
can be simplified with: https://github.com/crypto-com/chain/pull/554#discussion_r342399830

Reply:
it shouldn't be `+ 16_usize`, but `+` value between 1 and 15